Dockerfile

```diff
@@ -7,6 +7,8 @@ RUN set -x && \
          netcat python-pip virtualenv && \
     apt-get clean
 
+RUN pip install --upgrade setuptools
+
```

Comment:
AFAIK the `RUN` cmd gets cached by docker. Whereas `COPY, ADD` invalidate the cache, so every `RUN` cmd after those will get reevaluated every time the docker image is built, the `RUN` cmds before won't.

Having this in mind I can see that two different images can have two different versions of the setuptools installed leading to the error we saw on travis. But having this fix in place here could still lead to the same issue later, as it won't get reevaluated when rebuilding the image.
I therefore suggest to move this `RUN` cmd down after the first `COPY`cmd.